widget/embedded/src/hooks/useFetchQuote.ts

```diff
@@ -1,38 +1,51 @@
-import type { BestRouteRequest, BestRouteResponse } from 'rango-sdk';
+import type { RequestOptions } from 'https';
 
 import { useRef, useState } from 'react';
 
-import { httpService } from '../services/httpService';
-
-interface UseFetchQuote {
-  fetch: (requestBody: BestRouteRequest) => Promise<BestRouteResponse>;
+interface UseFetchQuoteOptions<TRequest, TResponse> {
+  endpoint: (
+    requestBody: TRequest,
+    options?: RequestOptions
+  ) => Promise<TResponse>;
+}
+export interface UseFetchQuoteResult<TRequest, TResponse> {
+  fetch: (requestBody: TRequest) => Promise<TResponse>;
```

Comment:
Prefer using the type keyword over interface to avoid declaration merging.

Reply:
fixed